src/server.rs

```diff
@@ -417,26 +417,46 @@ mod test {
 
     /// Attempts to grab a local, unbound socket address for testing.
     fn get_unbound_address() -> SocketAddr {
-        TcpListener::bind("127.0.0.1:0").unwrap().local_addr().unwrap()
+        let mut current = 4001;
+        loop {
+            let string = format!("127.0.0.1:{}", current);
+            let address = SocketAddr::from_str(&string).unwrap();
+            match TcpListener::bind(address) {
+                Ok(socket) => {
+                    info!("Next unbound address is: {:?}", socket.local_addr());
+                    return socket.local_addr().unwrap()
+                },
+                Err(_) => current += 1,
+            }
+        }
     }
 
     #[test]
+    // This should have an error because the peer address is the same as the test server address.
     pub fn test_illegal_peer_address() {
         let _ = env_logger::init();
         let peer_id = ServerId::from(1);
         let mut peers = HashMap::new();
-        peers.insert(peer_id, SocketAddr::from_str("127.0.0.1:0").unwrap());
-        assert!(new_test_server(peers).is_err());
+        peers.insert(peer_id, SocketAddr::from_str("127.0.0.1:4000").unwrap());
+        assert!(new_test_server(
+            ServerId::from(0),
+            SocketAddr::from_str("127.0.0.1:4000").unwrap(),
```

Comment:
This is a bad practice because tests will spuriously fail if the port is bound on the machine.